packages/gatsby/src/redux/nodes.ts

```diff
@@ -277,13 +329,17 @@ export const ensureIndexByElemMatch = (
       )
     })
   }
+
+  if (op === `$lte`) {
```

Comment:
Curious why `$lte` only? We can probably also support other range operators with a single index? Or is it something for a follow-up?

Reply:
Yes this is followup. I want to start with lte because that'll introduce this whole sorted list meta data thing. Other ops can reuse most of that work so we can put them in separate PRs.